controller/api/public/stat_summary.go

```diff
@@ -51,9 +49,15 @@ type podCount struct {
 
 func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest) (*pb.StatSummaryResponse, error) {
 	// special case to check for services as outbound only
-	if req.Selector.Resource.Type == k8s.Services &&
-		req.Outbound.(*pb.StatSummaryRequest_FromResource) == nil {
-		return nil, status.Errorf(codes.InvalidArgument, "service only supported as a target on 'from' queries, or as a destination on 'to' queries.")
+	if isInvalidServiceRequest(req) {
+		return &pb.StatSummaryResponse{
+			Response: &pb.StatSummaryResponse_Error{
+				Error: &pb.ResourceError{
```

Comment:
wow we were not using `ResourceError` at all!

Reply:
Yeeaaahhhh. We might want to consider using this in more places too, in follow-up changes.